enclave_manager/avalon_enclave_manager/wpe/wpe_requester.py

```diff
@@ -41,73 +46,157 @@ def __init__(self, config):
         self._uri_client = HttpJrpcClient(
             config["KMEListener"]["kme_listener_url"])
 
-    def get_unique_verification_key(self, nonce):
+    def get_unique_verification_key(self, verification_key_nonce,
+                                    encryption_key, verifying_key):
         """
         Request wrapper to get a unique id from the KME
 
         Parameters :
-            @param nonce - Random hex nonce generated by this WPE
+            @param verification_key_nonce - Random nonce generated by this WPE
+            @param encryption_key - Encryption key of worker to encrypt request
+            @param verifying_key - Verifying key of the worker
         Returns :
             @returns result - Result received from the KME which includes the
                               public verification key which is supposed to be
                               included in REPORTDATA by the WPE. None, in case
                               of failure.
         """
+
+        workload_id = "kme-uid"
+        in_data = json.dumps({"nonce": verification_key_nonce})
+
+        # Create session key and iv to sign work order request
+        session_key = crypto_utils.generate_key()
+        session_iv = crypto_utils.generate_iv()
+
+        wo_req = self._construct_wo_req(
+            in_data, workload_id, encryption_key, session_key, session_iv)
+
         json_rpc_request = self._get_request_json("GetUniqueVerificationKey")
-        json_rpc_request["params"] = {"nonce": nonce}
+        json_rpc_request["params"] = {"wo_request": wo_req}
         response = self._post_and_get_result(json_rpc_request)
 
         if "result" in response:
-            return response["result"]
+            wo_response_json = response["result"]
+            if self._verify_res_signature(wo_response_json, verifying_key):
+                decrypted_res = crypto_utils.decrypted_response(
+                    wo_response_json, session_key, session_iv)
+                # Response contains an array of results. In this case, the
+                # array has single element and the data field is of interest.
+                # The data contains result,verification_key and
+                # verification_key_signature delimited by ' '.
+                return decrypted_res[0]['data']
```

Comment:
Why didn't we use multiple index fields in outData to store verification_key and verification_key_signature ?

Reply:
There is a limitation of sending one outdata per indata field.

Reply:
I'm not aware of any such limitation in TCF. inData and outData is interpreted by client and workload